File: Models/GraphData.cs
using System;
using System.Collections.Generic;

namespace TeachingAidMac.Models
{
    [Serializable]
    public class GraphData
    {
        public List<NodeData> Nodes { get; set; } = new List<NodeData>();
        public string SourceNodeLetter { get; set; } = "";
        public string TargetNodeLetter { get; set; } = "";

        public void AddNode(NodeData addedNode)
        {
            Nodes.Add(addedNode);
        }

        public void SetSourceNodeLetter(string sourceNodeLetter)
        {
            SourceNodeLetter = sourceNodeLetter;
        }

        public void SetTargetNodeLetter(string targetNodeLetter)
        {
            TargetNodeLetter = targetNodeLetter;
        }

        public List<NodeData> GetGraphNodes()
        {
            return Nodes;
        }

        public void ClearGraph()
        {
            Nodes.Clear();
        }

        public string GetSourceNodeLetter()
        {
            return SourceNodeLetter;
        }

        public string GetTargetNodeLetter()
        {
            return TargetNodeLetter;
        }
    }
}

File: GraphData.vb
<Serializable> 
Public Class GraphData 

    Private Nodes As New List(Of NodeData) 

    Private SourceNodeLetter As String 

    Private TargetNodeLetter As String 

    Public Sub AddNode(AddedNode As NodeData) 

        Me.Nodes.Add(AddedNode) 

    End Sub 

    Public Sub SetSourceNodeLetter(SourceNodeLetter As String) 

        Me.SourceNodeLetter = SourceNodeLetter 

    End Sub 

    Public Sub SetTargetNodeLetter(TargetNodeLetter As String) 

        Me.TargetNodeLetter = TargetNodeLetter 

    End Sub 

    Public Function GetGraphNodes() As List(Of NodeData) 

        Return Me.Nodes 

    End Function 

    Public Sub ClearGraph() 

        Me.Nodes.Clear() 

    End Sub 

    Public Function GetSourceNodeLetter() As String 

        Return Me.SourceNodeLetter 

    End Function 

    Public Function GetTargetNodeLetter() As String 

        Return Me.TargetNodeLetter 

    End Function 

End Class 